### DISCUSSION TITLE
Is There A Way to Get Previous Version of Downloader with the Option to Create Series Folders? Or a way to sort the downloads?

### DISCUSSION THREAD

**Discussion by ghost** [nianeyna/ao3downloader#66] [Q&A] [answered]
Hello

@nianeyna Thank you so much for making this program! I constantly use it to download my favorites! I love it! Unfortunately I ran into a problem.

I got the new version of python and downloader as I've heard it is faster and only then noticed that the function to put works in series folders is gone. I was wondering if it might be possible to get a version of the downloader that still has that feature? I unfortunately deleted the previous version of the program and could not recover it.

The thing is that I favorited a lot of series and when I used the downloader they are in completely random spots in the folder and there is no way to sort them by author or series since now they are in a big group together. When the make series folders was available having that was very helpful. My strategy was basically to make a folder for fandom and then override those bookmarks whenever i updated them.

If it is not possible to get the previous version of the downloader is there some way of sorting the downloads? Maybe another program to do so? I can try to do it by hand, but I don't think that when I search my folder it finds everything with that prompt.

Thank you for reading!

**Comment by nianeyna** [comment, marked as answer]
There's absolutely another program that can sort your fics for you and it works way better than the series subfolders option*, as it will also allow you to sort and filter by other metadata as well (such as tags!). The only problem is that I've found it doesn't work well if you chose to download your fics in HTML or PDF format - you can still use it, but it won't do as good a job capturing metadata. If your fics are in HTML or PDF format, I recommend using ao3downloader's redownload function to switch your library to an ebook format first (epub is best).

Here's what you do:

- Install [calibre](https://calibre-ebook.com/)
- Put all of your files into a single folder with no subfolders
- Click the arrow next to the "Add books" button in calibre and choose "Add from folders and sub-folders"
- When prompted "Assume all files are multiple formats of the same book" select No
- Select the folder containing your files and let 'er rip. It will take some time to import depending on how many files you have
- You can do this again as many times as you want to add more files to your library

Lots of people like to customize their calibre installations with specific settings and plugins to optimize it for use with fanfic - you can look into that if you want, but I can't tell you much about it as I find the default configuration is pretty much sufficient for my needs. I think the only thing I changed was the auto-merge behavior (Add books -> Control the adding of books -> Adding actions).

*which I know people were using! and I really agonized over getting rid of it. but unfortunately it was causing the same fic to be downloaded multiple times - sometimes a really ludicrous number of times - which was not only slow but, I felt, pretty rude to ao3. I want ao3downloader to be as well-behaved and cooperative as possible; generating a bunch of essentially spam traffic goes against that. I hope you understand!

**Reply by ghost** [reply]
Thank you very much and sorry for such a late reply. I have been using Calibre. It has been working well for me.
